### ISSUE TITLE
User/CourseUser Public Profile

### ISSUE THREAD

**Issue by allenwq** [Coursemology/coursemology2#623]
Depends on #624 

~~### User Profile ###~~
~~User profile page should display the basic info of the user and the courses related to the user.~~

~~Since we have multiple instance, information like ___courses should be scoped to current instance___. We should not display the course information from other instances.~~

~~And, if a global admin visits the page, instances related to this user should also be displayed.~~

 
### Course User Profile ###

Course user profile page should display the achievements of the user and the skill tree.

As v1 does, I think we should not display level or exp in this page  ?

Would these make sense ? any suggestions ?

**Comment by weiqingtoh** [comment]
User profile - I think that is all right, however do we want to separate between 'completed' courses and 'active' ones?

Course User profile - I'm doing the user badge now, which might probably end up in the same place as v1 (above the sidebar). If experience pts and levels are there, I think the main profile can omit it. Not too much of an issue if we eventually want to change it anyway.

**Comment by allenwq** [comment]
@weiqingtoh Thanks ! Yah I think it makes sense to separate completed and active courses.

**Comment by weiqingtoh** [comment]
I'll do the Course User Profile!

**Comment by kxmbrian** [comment]
How about if we put the EXP history as a tab on the Course User profile page?
In v1, it is just titled "EXP History for XXX". It would be nicer to have the user badge as the header.

**Comment by allenwq** [comment]
@kxmbrian Yah, think tabs with user badges on top looks better.

**Comment by weiqingtoh** [comment]
Actually do we want to have the public profile for Users? 

What I see now is that this is implemented in v1 but I only found the page after some digging. It shows which course I've been in and some related stats (# achievements, # levels, role, etc.).

**Comment by benleong** [comment]
Yes. But not the most urgent feature at hand.

**Comment by benleong** [comment]
But can we quickly fix the profile photo uploading?

**Comment by weiqingtoh** [comment]
@benleong can elaborate a little? Currently students can upload their profile photo already.

**Comment by benleong** [comment]
Are we talking about the same thing?

**Comment by allenwq** [comment]
I will move user profile to a new issue and different milestone and close this.
